src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyRetryResponse.java

```diff
@@ -0,0 +1,143 @@
+import org.reaktivity.nukleus.http_cache.internal.types.stream.BeginFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.DataFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.EndFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.HttpBeginExFW;
+
+final class HttpCacheProxyRetryResponse
+{
+    private final HttpCacheProxyFactory httpCacheProxyFactory;
+    private final HttpCacheProxyCacheableRequest request;
+
+    private final int initialWindow;
+    private int connectReplyBudget;
+
+
+    HttpCacheProxyRetryResponse(
+        HttpCacheProxyFactory httpCacheProxyFactory,
+        HttpCacheProxyCacheableRequest request)
+    {
+        this.httpCacheProxyFactory = httpCacheProxyFactory;
+        this.request = request;
+        this.initialWindow = httpCacheProxyFactory.responseBufferPool.slotCapacity();
+    }
+
+    @Override
+    public String toString()
+    {
+        return String.format("%s[connectRouteId=%016x, connectReplyStreamId=%d]",
+                             getClass().getSimpleName(),
+                             request.connectRouteId,
+                             request.connectReplyId);
+    }
+
+    void onResponseMessage(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch (msgTypeId)
+        {
+            case BeginFW.TYPE_ID:
+                final BeginFW begin = httpCacheProxyFactory.beginRO.wrap(buffer, index, index + length);
+                onBegin(begin);
+                break;
+            case DataFW.TYPE_ID:
+                final DataFW data = httpCacheProxyFactory.dataRO.wrap(buffer, index, index + length);
+                onData(data);
+                break;
+            case EndFW.TYPE_ID:
+                final EndFW end = httpCacheProxyFactory.endRO.wrap(buffer, index, index + length);
+                onEnd(end);
+                break;
+            case AbortFW.TYPE_ID:
+                final AbortFW abort = httpCacheProxyFactory.abortRO.wrap(buffer, index, index + length);
+                onAbort(abort);
+                break;
+        }
+    }
+
+    private void onBegin(
+        BeginFW begin)
+    {
+        final long connectReplyId = begin.streamId();
+        final OctetsFW extension = httpCacheProxyFactory.beginRO.extension();
+        final HttpBeginExFW httpBeginFW = extension.get(httpCacheProxyFactory.httpBeginExRO::wrap);
+
+        if (DEBUG)
+        {
+            System.out.printf("[%016x] CONNECT %016x %s [received response]\n", currentTimeMillis(), connectReplyId,
+                              getHeader(httpBeginFW.headers(), ":status"));
+        }
+
+        final ListFW<HttpHeaderFW> responseHeaders = httpBeginFW.headers();
+        String status = getHeader(responseHeaders, STATUS);
+        long retryAfter = HttpHeadersUtil.retryAfter(responseHeaders);
+        assert status != null;
+
+        request.scheduleRequest(retryAfter);
```

Comment:
Defer scheduling next request until after previous response has completed in either `onEnd` or `onAbort`.

Reply:
Done.